### PR TITLE
Improved Map/Set usage

### PR DESCRIPTION
**Pull request by tbw777**
- keySet().contains() -> containsKey()
- values().contains() -> containsValue()
- keySet() -> entrySet() - both keys and values are used
- keySet() -> values() - keys used only with get()
- entrySet() -> values() - keys not used
- entrySet() -> keySet() - values not used
- remove() -> removeAll() - applicable
- put() -> putAll()

Targets: performance, structure, clean code

### CODE REVIEW

**Comment by mbien** [comment]
as previously mentioned by several committers in #5238. I would not recommend to create more PRs since you will run into merge conflicts and cause other problems later.

Running automated refactorings is easy and fast, reviewing the changeset and making sure that:
 - nothing breaks
 - no other PRs are negatively affected by project wide changes, esp during RC phase when the master and delivery branches should not diverge unnecessarily since this would make PRs no longer portable
 - and CI is not flooded,

is not trivial and also time consuming.

Since you have been sufficiently warned before I will be simply closing new cleanup PRs from now on. Your contributions are appreciated but cleanups have to be done in a way which does not disturb the project itself.

cc @matthiasblaesing @neilcsmith-net @BradWalker

**Comment by tbw777** [comment]
its not a problem to rebase to latest master any my branch by my hands )

**Comment by BradWalker** [comment]
The large number of files is not a small amount of work to review. I've done a lot of "simple" code cleanup in the past. My work was clearly tied to the removal of warning messages.  These changes don't seem to really be tied to a problem or issue. They appear to me to be code clean just for the sake of code cleanup. 

When I did code clean up in the past here are some reminders that I was always telling myself:
1 - is this work too large to be easily reviewed, if so break it down
2 - i always tried to pace myself, that way i wasn't overwhelming the team with code reviews
3 - i often had a a review ready to go but did not submit it because wanted to wait for the previous one to be approved - again pacing myself
4 - i tried to listen to the feedback that i was getting, both positive and negative

It's always a good thing to ask oneself "what am I trying to accomplish"..

**Comment by tbw777** [comment]
@mbien dear, can i do PR's that removed some compile warnings or you also skip it?
Can i do migrate modules step by step to gradle?

**Comment by matthiasblaesing** [comment]
>  Can i do migrate modules step by step to gradle?

Short: No. Long: Changing the build system is a matter for all developers and needs to be discusses on the mailing list. There are multiple issues (for example there are people here, that think gradle is not a great build system) and there are already people out there, that tried to convert the build to gradle and have not yet succeeded. As said this is a major issue and at this point in time I'd veto any PR, that tries to introduce a mix of build systems.

> dear, can i do PR's that removed some compile warnings or you also skip it?

Not sure what you try to say. I can't speak for "we", but for me: I will look at PRs in the pace that I can manage, while also progressing "my" pet issues.  You can ask on the mailing list for more committers to look.

At this point in time PRs that touch many file are a bad idea, as currently the release for 17 is being prepared (as announced on the mailing list).

> Can i make supercommit with replacing " + " into "append" change there SB already used? 

If you mean you want to change this:

```java
StringBuilder sb = new StringBuilder();
sb.append("Hallo" + "Welt");
```

to

```java
StringBuilder sb = new StringBuilder();
sb.append("Hallo");
sb.append("Welt");
```

Sure. But remember: You need to motivate people to review. In my mind, if the code is not performance critical, the first form is not optimal, but works and thus does not need to change. If you fix a real bug and change it then, it is a different matter.

All this is IMHO and others might feel differently.

**Comment by neilcsmith-net** [comment]
> for example there are people here, that think gradle is not a great build system

Gradle is just Ant with slightly better syntax! :smile:  In seriousness, changing the build system for the sake of it is probably not a good use of anyone's time.  Firstly quantify the benefits on the dev@ list.

> In my mind, if the code is not performance critical, the first form is not optimal, but works and thus does not need to change.

And given JEP 280, if it was performance critical then switching to solely concatenation might be better anyway.

**Comment by tbw777** [comment]
> ```java
> sb.append("Hallo");
> sb.append("Welt");
> ```

no, this way:
> ```java
> sb.append("Hallo").append("Welt");
> ```

But as I remembered that jep 280 really can fix this and we dont needed to change anything not in cycles. Only cycles need to be updated. I must to make some tests. IMHO the right way will be to update bytecode version of the project. What do you think?

File: platform/core.windows/src/org/netbeans/core/windows/view/ViewHierarchy.java
Comment (mbien) [inline platform/core.windows/src/org/netbeans/core/windows/view/ViewHierarchy.java:1003]: revert `append()` change. `+` is fine.

File: platform/openide.util/test/unit/src/org/openide/util/NbCollectionsTest.java
Comment (mbien) [inline platform/openide.util/test/unit/src/org/openide/util/NbCollectionsTest.java:311]: this is a NbCollectionsTest! Lets revert this.

File: webcommon/javascript2.editor/test/unit/src/org/netbeans/modules/javascript2/editor/formatter/JsFormatterTestBase.java
Comment (mbien) [inline webcommon/javascript2.editor/test/unit/src/org/netbeans/modules/javascript2/editor/formatter/JsFormatterTestBase.java:143]: rename to `option` like it was before.

File: webcommon/javascript2.editor/test/unit/src/org/netbeans/modules/javascript2/editor/formatter/PackageFormatterTest.java
Comment (mbien) [inline webcommon/javascript2.editor/test/unit/src/org/netbeans/modules/javascript2/editor/formatter/PackageFormatterTest.java:113]: same here -> rename to option

File: webcommon/lib.v8debug/src/org/netbeans/lib/v8debug/client/cmdline/V8Debug.java
Comment (mbien) [inline webcommon/lib.v8debug/src/org/netbeans/lib/v8debug/client/cmdline/V8Debug.java:1026]: use `propName` variable for key like it was before

**Review by mbien — CHANGES_REQUESTED** [review]
Overall good changes, thanks. Some minor requests inline.

File: webcommon/lib.v8debug/src/org/netbeans/lib/v8debug/client/cmdline/V8Debug.java
Comment (mbien) [inline webcommon/lib.v8debug/src/org/netbeans/lib/v8debug/client/cmdline/V8Debug.java:1026]: this PR is not about `append`. Please keep the code as is and restrict changes to collections.

File: apisupport/timers/src/org/netbeans/modules/timers/TimeComponentPanel.java
Comment (mbien) [inline apisupport/timers/src/org/netbeans/modules/timers/TimeComponentPanel.java:449]: here is another change to append. I think it is the last one.

**Comment by mbien** [comment]
@tbw777 also, can you please update the commit message to the PR title, so that it is less vague.

File: enterprise/j2ee.sun.ddui/src/org/netbeans/modules/j2ee/sun/ddloaders/multiview/webservice/WebServiceMetadataReader.java
Comment (mbien) [inline enterprise/j2ee.sun.ddui/src/org/netbeans/modules/j2ee/sun/ddloaders/multiview/webservice/WebServiceMetadataReader.java:94]: service name are already distinct. This can be reverted back to a list.

File: apisupport/timers/src/org/netbeans/modules/timers/TimeComponentPanel.java
Comment (mbien) [reply apisupport/timers/src/org/netbeans/modules/timers/TimeComponentPanel.java:449]: @tbw777 you forgot this one. Please revert the change to append.

File: enterprise/websvc.core/src/org/netbeans/modules/websvc/core/jaxws/projects/JavaEEWSOpenHook.java
Comment (mbien) [reply enterprise/websvc.core/src/org/netbeans/modules/websvc/core/jaxws/projects/JavaEEWSOpenHook.java:221]: @tbw777 you marked this as resolved but forgot to revert.

File: ide/project.ant/src/org/netbeans/spi/project/support/ant/ReferenceHelper.java
Comment (mbien) [inline ide/project.ant/src/org/netbeans/spi/project/support/ant/ReferenceHelper.java:1351]: revert please. Elements are already distinct so they can be stored in a list.

**Review by mbien — COMMENTED** [review]
took another look. A few more things need to be reverted. Two which you forgot from the last time.

**Comment by tbw777** [comment]
Set was introduced to increase readability and for removeAll()/contains()

**Comment by mbien** [comment]
> Set was introduced to increase readability and for removeAll()/contains() calls. Reverted

NB code didn't call `contains()` and taking a look at the `removeAll()` impl:
https://github.com/openjdk/jdk/blob/1fec6b5953b51dae4be640d6e4e4f79136b9348d/src/java.base/share/classes/java/util/AbstractSet.java#L166-L182

`contains()` is only called if the `toRemove` collection is larger than the set, however this can't happen as far as I see since `toRemove` is built by going through the collection first - so it is always smaller.

I do agree though that calling `set.removeAll(veryLargeList)` would potentially scale badly. But in the concrete cases here where we know how the list is build it should not cause this problem.

**Review by mbien — APPROVED** [review]
all tests green. Changes look good to me now.

@matthiasblaesing @errael @lkishalmi I am going to leave this open for a few days to give time for a second review before merge. Should be easier now since the changeset is smaller.

File: platform/options.keymap/src/org/netbeans/modules/options/keymap/KeymapPanel.java
Comment (errael) [inline platform/options.keymap/src/org/netbeans/modules/options/keymap/KeymapPanel.java:304]: Almost done, I'm getting punchy... Again not looking for a change, just mentioning, a check on myself

```
for(List<String> categories : getModel().getCategories().values()) {
    for(String category : categories) {
```

File: platform/options.keymap/src/org/netbeans/modules/options/keymap/KeymapViewModel.java
Comment (errael) [inline platform/options.keymap/src/org/netbeans/modules/options/keymap/KeymapViewModel.java:234]: Ditto

**Review by errael — APPROVED** [review]
Looks OK to me.

> Should be easier now since the changeset is smaller

Wow

**Comment by mbien** [comment]
@errael thanks for the review (and everyone else), I looked a second time through it and I believe it is good now. This one was time consuming - lets hope it was worth the effort.

all tests green again -> lets get this in

File: ide/bugtracking/src/org/netbeans/modules/bugtracking/RepositoryImpl.java
Comment (errael) [reply ide/bugtracking/src/org/netbeans/modules/bugtracking/RepositoryImpl.java]: > #5464

@tbw777 Not sure what this reference is for. 

And there was a bit of miscommunication. From what I've read, using `ArrayList` is almost always a win over using `LinkedList`. I'm not familiar with this code, it's unlikely to be an issue either way in this case.

Comment (mbien) [reply ide/bugtracking/src/org/netbeans/modules/bugtracking/RepositoryImpl.java]: i think NB code uses LL in places where there are only a handful of elements expected. But I do agree that ALs are basically the right choice in 99% of all cases, they create fewer objects and if you copy things around AL will always win.

Some edge cases like insert-in-front can cause trouble though.

**Pull request merged**
